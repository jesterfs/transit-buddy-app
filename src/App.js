import React, { Component } from 'react';
import {Route} from 'react-router-dom';
import LandingNav from './landing/landing-nav'
import LandingBody from './landing/landing-body.js'
import Footer from './footer/footer'
import LoginBody from './login/login-body';
import LoginNav from './login/login-nav';
import SignupNav from './signup/signup-nav';
import SignupBody from './signup/signup-body';
import DashboardNav from './dashboard/dashboard-nav';
import DashboardBody from './dashboard/dashboard-body';
import ApiContext from './ApiContext';
import store from './store';
import cfg from './config.js'
import TokenServices from './services/token-services'
import { Fade } from 'react-awesome-reveal'

class App extends Component {

  state = {
    user: null,
    // members: store.members
};

signup = (user) => {
  this.setState({
    user: user,
    members: [...this.state.members, user]
  })
  
  
}



changeUser = (user) => {
  this.setState({user: user })
}

fetchUserData = (id) => {
  fetch(cfg.API_ENDPOINT + `members/${id}`, {
    method: 'GET', 
    headers: {
      'Authentication' : `Bearer ${TokenServices.getAuthToken()}`,
      'Content-Type': 'application/json',
    }
  })
      .then(response => response.json())
      .then(data => 
        this.setState({
          user: data
        }, 
      ))
}






componentDidMount() {
  const info = TokenServices.getAuthInfo(); 
  if(info) {
    this.fetchUserData(info.userId) 
  }
  
}

renderNavRoutes() {
  return(
    <>
    <Route exact path="/" component={LandingNav} />
    <Route exact path="/login" component={LoginNav} />
    <Route exact path="/signup" component={SignupNav} />
    <Route exact path="/dashboard" component={DashboardNav} />
    </>
  )
}

  // renderGreetingRoutes() {
  //   return(
      
  //   )
  // }

  renderBodyRoutes() {
    return(
        <>
        <Route exact path="/" component={LandingBody} />
        <Route 
            exact path="/login" 
            component={LoginBody}
            />
        <Route exact path="/signup" component={SignupBody} />
        <Route exact path="/dashboard" component={DashboardBody} />
        </>
    )
  }
  
  renderFooterRoutes() {
    return(
      <>
      <Route path="/" component={Footer} />
      </>
    )
  }


  
  render() {
    const value = {
      user: this.state.user,
      members: this.state.members,
      changeUser: this.changeUser,
      signup: this.signup,
      fetchUserData: this.fetchUserData
    }
    return (
      <ApiContext.Provider value={value}>
        <div className='App'>
          <Fade triggerOnce>
          <nav>
            {this.renderNavRoutes()}
          </nav>
          <main className='App'>
            
            {this.renderBodyRoutes()}
          </main>
          <footer>
            {this.renderFooterRoutes()}
          </footer>
          </Fade>
        </div>
      </ApiContext.Provider>
        
      
      
    );
  }
  
}

export default App;

