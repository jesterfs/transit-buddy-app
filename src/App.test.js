import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import App from './App'
import renderer from 'react-test-renderer';

describe('App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders the UI as expected', () => {
    const div = document.createElement('div')
    const tree = renderer
      .create(<BrowserRouter>
        <App />
      </BrowserRouter>,
      div)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

});



