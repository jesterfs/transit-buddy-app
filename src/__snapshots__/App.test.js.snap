// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the UI as expected 1`] = `
<div
  className="App"
>
  <div
    className="css-q7lffx"
  >
    <nav
      style={Object {}}
    >
      <div
        className="LandingNav, group"
      >
        <div
          className="item navbtn"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Log In
          </a>
        </div>
        <div
          className="item"
        >
          <a
            href="/"
            onClick={[Function]}
          >
             
            <img
              alt="Transit Buddy"
              src="logo.png"
            />
             
          </a>
        </div>
        <div
          className="item"
          id="signupbtn"
        >
          <a
            href="/signup"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="css-q7lffx"
  >
    <main
      className="App"
      style={Object {}}
    >
      <div
        className="landingbody"
      >
        <div
          className="landinggroup group1"
        >
          <div
            className="item"
          >
            <img
              alt="TransitBuddy Dashboard"
              className="landing-img desktop"
              src="screenshot.png"
            />
            <img
              alt="TransitBuddy Dashboard"
              className="landing-img mobile"
              src="mobile-screenshot.png"
            />
          </div>
          <div
            className="item"
          >
            <h3>
              Riders Helping Riders
            </h3>
            <p>
              Navigating complicated public transit systems is hard enough without dealing with construction, out of service elevators, and other obstacles that get in your way. With Transit Buddy, you can make your commute easier and help out your fellow riders along the way. We let you report issues along your route and see what problems other passengers have run into.
            </p>
          </div>
        </div>
        <div
          className="landinggroup2"
        >
          <div
            className="item"
          >
            <h3>
              Create Your Account Today
            </h3>
            <a
              href="/signup"
              onClick={[Function]}
            >
              <button>
                Sign Up
              </button>
            </a>
          </div>
        </div>
      </div>
    </main>
  </div>
  <div
    className="css-q7lffx"
  >
    <footer
      style={Object {}}
    >
      <div
        className="footergroup"
      >
        <div
          className="item"
        >
          <p>
            Copyright 2020 by Franklin Stewart Jester
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
